lib/capistrano/doctor/servers_doctor.rb

```diff
@@ -0,0 +1,70 @@
+require "capistrano/doctor/output_helpers"
+
+module Capistrano
+  module Doctor
+    class ServersDoctor
+      include Capistrano::Doctor::OutputHelpers
+
+      def initialize(env=Capistrano::Configuration.env)
+        @servers = env.servers.to_a
+      end
+
+      def call
+        title("Servers (#{servers.size})")
+
+        table(servers) do |server, row|
+          sd = ServerDecorator.new(server)
+          row << sd.uri_form
+          row << sd.roles
+          row << sd.properties
+        end
+
+        puts
+      end
+
+      private
+
+      attr_reader :servers
+
+      class ServerDecorator
+        def initialize(server)
+          @server = server
+        end
+
+        def uri_form
+          [
+            server.user,
+            server.user && "@",
+            server.hostname,
+            server.port && ":",
```

Comment:
I really like this `value && separator` shorthand you used here in combination with `.compact.join`. Elegant!